### PR TITLE
GO-3849 Fix IsEmpty for sets

### PR DESCRIPTION
**Pull request by KirillSto**
https://linear.app/anytype/issue/GO-3849/need-to-save-the-setcollection-if-you-have-not-entered-a-name

DeleteIfEmpty internal flag is unset on state.IsEmpty returning false. However this check analyzes only text block contents.
Sets and collections have only standard blocks, so separate logic to check object emptiness must be supported

### CODE REVIEW

File: core/block/editor/state/state.go [hunk 1]
```diff
@@ -1378,6 +1385,35 @@ func (s *State) IsEmpty(checkTitle bool) bool {
 	return true
 }
 
+// isEmptyDataview is a separate emptiness check for sets and collections, as
```
Comment (AnastasiaShemyakinskaya) [inline core/block/editor/state/state.go:1388]: as.....?🤔

File: core/block/simple/dataview/dataview.go [hunk 2]
```diff
@@ -520,3 +522,29 @@ func (s *Dataview) UpdateRelationOld(relationKey string, rel model.Relation) err
...
+	case 1:
+		view = d.content.Views[0]
+	default:
+		return false
+	}
+
+	if view.DefaultObjectTypeId != "" ||
+		view.DefaultTemplateId != "" ||
+		len(view.Filters) != 0 ||
+		len(view.Sorts) > 1 ||
```
Comment (AnastasiaShemyakinskaya) [inline core/block/simple/dataview/dataview.go:544]: Why do we check `len(view.Sorts)` as more then 1? Just want to understand the logic 😃

File: core/block/editor/state/state.go (hunk 1 above)
Comment (KirillSto) [reply core/block/editor/state/state.go:1388]: Sorry, started typing and did not finish :smile:

File: core/block/simple/dataview/dataview.go (hunk 2 above)
Comment (KirillSto) [reply core/block/simple/dataview/dataview.go:544]: On creation of new dataview block we add sorting by lastModified by default:
https://github.com/anyproto/anytype-heart/blob/main/core/block/editor/dataview/dataview.go#L528
so if we find some other sorts in the view, we can consider dataview object as modified

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Pull request closed without merging**